### PR TITLE
Seasonality check Flag

### PR DESCRIPTION
**Pull request by TremaMiguel**
#### Describe the changes you've made

Check seasonal period in data through sktime's `autocorrelation_seasonality_test` function.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

No tests need.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by Yard1 — APPROVED** [review]
LGTM

File: pycaret/internal/pycaret_experiment/time_series_experiment.py [hunk 1]
```diff
@@ -368,12 +370,14 @@ def setup(
...
             if isinstance(seasonal_parameter, str):
                 try:
-                    seasonal_parameter = SeasonalParameter[seasonal_parameter]
+                    self.seasonal_parameter = SeasonalParameter[seasonal_parameter]
                 except KeyError:
                     raise ValueError(
                         f"Unsupported Period frequency: {seasonal_parameter}, valid Period frequencies: {', '.join(SeasonalParameter.__members__.keys())}"
                     )
 
+            self.seasonal_parameter = seasonal_parameter
```
[the remark this answers is not in the document]
Comment (TremaMiguel) [reply pycaret/internal/pycaret_experiment/time_series_experiment.py:379]: yes, I was missing assigning this case when `seasonal_parameter` is integer.

[the remark this answers is not in the document]
Comment (ngupta23) [reply pycaret/internal/pycaret_experiment/time_series_experiment.py]: Thanks @TremaMiguel , Can you add unit tests to test for these 2 cases (type of the argument)?

[the remark this answers is not in the document]
Comment (TremaMiguel) [reply pycaret/internal/pycaret_experiment/time_series_experiment.py]: you mean testing the exceptions raised `seasonal_parameter parameter must be an int or str` and `Unsupported Period frequency` ?

[the remark this answers is not in the document]
Comment (ngupta23) [reply pycaret/internal/pycaret_experiment/time_series_experiment.py]: Yes, that could be one of them. In addition, I was referring to specifically testing passing of int and str values and checking that the internal values are set correctly.

**Review by ngupta23 — APPROVED** [review]

**Pull request merged**